java-client/src/main/java/org/opensearch/client/json/JsonpSerializable.java

```diff
@@ -33,11 +33,33 @@
 package org.opensearch.client.json;
 
 import jakarta.json.stream.JsonGenerator;
+import org.opensearch.client.json.jackson.JacksonJsonProvider;
+import org.opensearch.client.json.jackson.JacksonJsonpMapper;
+
+import java.io.StringWriter;
 
 /**
  * An object that is its own JsonP serializer
  */
 public interface JsonpSerializable {
 
     void serialize(JsonGenerator generator, JsonpMapper mapper);
+
+
+    /**
+     * A default method which returns string representation for the instances of classes
+     * implementing JsonpSerializable interface.<br>
+     * Usage : Eg for SearchRequest.class<pre>{@code SearchRequest implements JsonpSerializable{..}
+     * SearchRequest searchRequest = SearchRequest.of(request -> request...);
+     * String searchRequestString = searchRequest.writeValueAsString();}</pre>  <br>
+     *
+     */
+    default String writeValueAsString() {
+        StringWriter writer = new StringWriter();
+        try (JsonGenerator generator = JacksonJsonProvider.provider().createGenerator(writer)) {
```

Comment:
This is not going to work (since Jackson is not required), could you please port https://github.com/opensearch-project/spring-data-opensearch/blob/main/spring-data-opensearch/src/main/java/org/opensearch/data/client/osc/JsonpUtils.java - the idea there is to use the provider from the classpath (either Jakarta or Jackson) but not hardcode any/

Reply:
Ohh got it, ported default provider and mapper from [PR](https://github.com/opensearch-project/spring-data-opensearch/blob/main/spring-data-opensearch/src/main/java/org/opensearch/data/client/osc/JsonpUtils.java), rather than hardcoding the service providers.